barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3844,12 +4142,16 @@ std::vector<Row> AvmTraceBuilder::finalize()
     for (size_t i = 0; i < new_trace_size; i++) {
         auto& r = main_trace.at(i);
 
+        if (r.main_tag_err == FF(1)) {
+            r.main_op_err = FF(1); // Consolidation of errors into main_op_err
+        }
+
         if ((r.main_sel_op_add == FF(1) || r.main_sel_op_sub == FF(1) || r.main_sel_op_mul == FF(1) ||
```

Comment:
When we constrain yes. The tag_err is very specific and it is easier to reason about to isolate from other potential errors. I think it would be diificult to use a single error flag. In addition, we have specific constraints related to tag_err that I had to use op_err for the non-constrained tag errors, otherwise we would have wrong constraints.